library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/CarrierDoctrineRepository.php

```diff
@@ -69,4 +71,50 @@ public function getCarrierIdsByBrandAdmin(AdministratorInterface $admin): array
             'id'
         );
     }
+
+    /**
+     * @param BrandInterface $brand
+     * @param ProxyTrunkInterface $proxyTrunks
+     * @return array|CarrierInterface[]
+     */
+    public function findByBrandAndProxyTrunks(BrandInterface $brand, ProxyTrunkInterface $proxyTrunks)
+    {
+        /** @var CarrierInterface[] $carriers */
+        $carriers = $this->findAll();
+        $response = [];
+
+        foreach ($carriers as $carrier) {
+            if ($carrier->getBrand() !== $brand) {
+                continue;
+            }
+            if ($carrier->getProxyTrunk() !== $proxyTrunks) {
+                continue;
+            }
+
+            $response[] = $carrier;
+        }
+
+        return $response;
```

Comment:
Maybe I dont understand this logic, but you're trying to search from Carriers those that have both that brand and proxyTrunk assigned? You can get that directly without looping carriers.

```suggestion
        $response = $this->findBy([
           'brand' => $brand,
           'proxyTrunk' => $proxyTrunks
        ]);
        
        return $response;
```